server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorFactory.java

```diff
@@ -50,6 +54,14 @@ static void registerAggregators(ValuesSourceRegistry.Builder builder) {
             TermsAggregatorFactory.numericSupplier(), true);
     }
 
+    /**
+     * The maximum number of global ordinals a field can have for us to try
+     * aggregating it "filter by filter". "Filter by filter" aggregation is
+     * generally faster when possible but takes more memory because we have
+     * to build the filters.
+     */
+    static final long MAX_ORDS_TO_TRY_FILTERS = 1000;
```

Comment:
I've just pulled this number out of a hat. My wild guess is we spend 300 bytes on each ord to use `filters`. I'll bet we could make `filters` use less memory when we're building it internally. But maybe we bump it then?